mm2src/coins/z_coin/z_coin_native_tests.rs

```diff
@@ -1,143 +1,155 @@
-use super::*;
-use crate::z_coin::z_htlc::z_send_dex_fee;
-use common::block_on;
-use common::now_ms;
+use bitcrypto::dhash160;
+use common::{block_on, now_ms};
 use mm2_core::mm_ctx::MmCtxBuilder;
+use mm2_test_helpers::for_tests::zombie_conf;
+use std::path::PathBuf;
 use std::time::Duration;
 use zcash_client_backend::encoding::decode_extended_spending_key;
 
+use super::{z_coin_from_conf_and_params_with_z_key, z_mainnet_constants, Future, PrivKeyBuildPolicy,
+            RefundPaymentArgs, SendPaymentArgs, SpendPaymentArgs, SwapOps, ValidateFeeArgs, ValidatePaymentError,
+            ZTransaction};
+use crate::z_coin::{z_htlc::z_send_dex_fee, ZcoinActivationParams, ZcoinRpcMode};
+use crate::{CoinProtocol, IguanaPrivKey};
+
 #[test]
 fn zombie_coin_send_and_refund_maker_payment() {
-    let conf = json!({
-        "coin": "ZOMBIE",
-        "asset": "ZOMBIE",
-        "fname": "ZOMBIE (TESTCOIN)",
-        "txversion": 4,
-        "overwintered": 1,
-        "mm2": 1,
-    });
-    let req = json!({
-        "method": "enable",
-        "coin": "ZOMBIE"
-    });
-
     let ctx = MmCtxBuilder::default().into_mm_arc();
-    let priv_key = [1; 32];
+    let mut conf = zombie_conf();
+    let params = default_zcoin_activation_params();
+    let priv_key = PrivKeyBuildPolicy::IguanaPrivKey(IguanaPrivKey::default());
+    let db_dir = PathBuf::from("./for_tests");
     let z_key = decode_extended_spending_key(z_mainnet_constants::HRP_SAPLING_EXTENDED_SPENDING_KEY, "secret-extended-key-main1q0k2ga2cqqqqpq8m8j6yl0say83cagrqp53zqz54w38ezs8ly9ly5ptamqwfpq85u87w0df4k8t2lwyde3n9v0gcr69nu4ryv60t0kfcsvkr8h83skwqex2nf0vr32794fmzk89cpmjptzc22lgu5wfhhp8lgf3f5vn2l3sge0udvxnm95k6dtxj2jwlfyccnum7nz297ecyhmd5ph526pxndww0rqq0qly84l635mec0x4yedf95hzn6kcgq8yxts26k98j9g32kjc8y83fe").unwrap().unwrap();
+    let CoinProtocol::ZHTLC(protocol_info) = serde_json::from_value::<CoinProtocol>(conf["protocol"].take()).unwrap() else { unimplemented!() };
```

Comment:
done

Reply:
For this panic
https://github.com/KomodoPlatform/atomicDEX-API/blob/2340ad8c2d574f183cfb5f06444e19a6cde7a19b/mm2src/coins/z_coin/z_coin_native_tests.rs#L25
No need to mention `zombie_conf` since this will can happen if a dev editing this test used another conf by mistake, it's also a wrong protocol not a failure to get protocol.
Again not a blocker :)

Reply:
done